app/Services/ViewOffersService.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Services;
+
+use App\Http\Resources\ErrorResource;
+use App\Http\Resources\SuccessResource;
+use App\Models\Offer;
+use App\Repositories\OfferRepository;
+use Exception;
+use Illuminate\Http\Response;
+
+class ViewOffersService
+{
+    protected $offerRepository;
+
+    public function __construct(OfferRepository $offerRepository)
+    {
+        $this->offerRepository = $offerRepository;
+    }
+
+    public function execute($request)
+    {
+        try {
+            $userId = $request->user()->id;
+            $paginate = $request->query("paginate");
+            $count = null;
+            if ($paginate) {
+                $count = $paginate;
+            }
+            $offers = $this->offerRepository->getNewOffers($userId, $request->query("category_id"))->paginate($count);
```

Comment:
@RanaEmad Can we create a request for validating category_id if it exists into the categories table and paginate is an integer

Reply:
These are supposed to be handled internally with a select box, the user will not enter them manually.